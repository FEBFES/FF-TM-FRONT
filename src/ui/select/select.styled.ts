import styled from 'styled-components'

export const SSelect = styled.select<{onlyView: boolean}>`
  font-size: 12px !important;
  color: var(--font-gray);
  background: transparent;
  border: 1px solid var(--bg-border);
  border-radius: 6px;
  text-transform: lowercase;
  padding: 6px 10px;
  outline: none;
  cursor: ${({onlyView}) => onlyView && 'not-allowed'}

  & :hover {
    outline: 1px solid var(--bg-primary);
  }
`

export const SOption = styled.option`
  font-size: 12px;
  color: var(--font-gray);
  padding: 4px 8px;
`