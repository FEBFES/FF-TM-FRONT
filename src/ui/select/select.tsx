import React from 'react';
import { SSelect, SOption } from './select.styled';

interface SelectProps {
  defaultValue: string;
  optionsArr: string[];
  onlyView?: boolean;
  onChange: Function;
}

export const Select: React.FC<SelectProps> = ({
  defaultValue,
  optionsArr,
  onlyView = false,
  onChange,
}): JSX.Element => {
  const changeHandler = (e: any) => {
    e.preventDefault();
    onChange(e.target.value);
  };

  return (
    <SSelect
      disabled={onlyView}
      onlyView={onlyView}
      defaultValue={defaultValue}
      onChange={changeHandler}
    >
      {optionsArr.map((option: string, index: number) => {
        return (
          <SOption key={`${option}-${index}`} value={option}>
            {option}
          </SOption>
        );
      })}
    </SSelect>
  );
};
