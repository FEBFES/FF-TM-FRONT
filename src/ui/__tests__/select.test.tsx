import React from 'react';
import { Select } from '../select/select';
import renderer from 'react-test-renderer';

it('SNAPSHOTS: UI - Select component - render', () => {
  const elem = renderer
    .create(
      <Select optionsArr={['test', 'test1', 'test2']} defaultValue="test" />
    )
    .toJSON();
  expect(elem).toMatchSnapshot();
});
