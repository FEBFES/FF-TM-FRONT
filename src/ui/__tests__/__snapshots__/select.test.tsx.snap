// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SNAPSHOTS: UI - Select component - render 1`] = `
<select
  className="sc-bcPKhP ffGrLR"
  defaultValue="test"
  disabled={false}
  onChange={[Function]}
  onClick={[Function]}
  onlyView={false}
>
  <option
    className="sc-grXZZQ cBhrEl"
    value="test"
  >
    test
  </option>
  <option
    className="sc-grXZZQ cBhrEl"
    value="test1"
  >
    test1
  </option>
  <option
    className="sc-grXZZQ cBhrEl"
    value="test2"
  >
    test2
  </option>
</select>
`;
